src/js/auth.js

```diff
@@ -14,49 +14,53 @@
  * limitations under the License.
  */
 
-/* global gapi */
-
 // Utility methods for dealing with Google auth:
-// https://developers.google.com/identity/sign-in/web/reference
+// https://developers.google.com/identity/oauth2/web/guides/use-token-model
 
-export async function checkAuthStatus() {
-  // This is set on the window in `index.html`.
-  await window.authorizeUser;
+window.tokenPromise = new Promise((resolve, reject) => {
+  window.tokenLoadOkay = resolve;
+  window.tokenLoadFail = reject;
```

Comment:
randomly came across here and got a little curious myself. 

found some official docs that have a few patterns: https://developers.google.com/identity/oauth2/web/guides/migration-to-gis#the_new_way  (the 3 tabs under The New Way are all small variants)

In the GAPI callback one they immediately do `initTokenClient` in the onload handler of the `gis` script. however they leave callback "undefined" and then redefine it later prior to requestAccessToken. I'm too lazy to understand our constraints here, but... just offering that in case it's interesting to you. :) 

------

code wise the overall approach here seems fine. 

for this one-shot promise there's a pattern i've used before..  https://github.com/GoogleChrome/lighthouse/pull/13966#discussion_r896091907  At first it looks weird, but you get used to it. It'd allow you to drop the globals.  

(again, this is all up to you guys. i'm dropping in out of the blue)

Reply:
This was only needed as was trying to load at start up, so needed to know when gis script was loaded.

Now we wait for user action to click on login link/button, so will just assume it's loaded by then and remove this.